x-pack/plugins/monitoring/server/lib/alerts/create_dataset_query_filter.ts

```diff
@@ -6,7 +6,15 @@
  */
 
 /**
- * metricset and dataset should match but there are currently exceptions which could just be bugs
+ * We expected metricset and dataset to match but it appears to be the case that
+ * dataset is the full {product}.{metricset}, whereas metricset doesn't include
+ * the product, e.g. dataset is elasticsearch.cluster_stats and metricset is
+ * just cluster_stats.
```

Comment:
sorry, that's not actually what i was referring to.  i was referring to the fact that the 2nd part of the datas_stream.dataset name does not always match metric.set name.  See comment https://github.com/elastic/kibana/pull/119112/files#r772605936 .  I don't know if this was a bug or if its been fixed.  @sayden said it was possibly a bug when I inquired about it.  This is why I separated them instead of concatenating because there were inconsistencies.

Reply:
Ah ok thanks for clarifying, I'll update the comment!

Reply:
Updated, let me know if that's still inaccurate. Thanks!

Reply:
Looks good, thanks!